### ISSUE TITLE
SWIG_exception writes warning first and then calls error()

### ISSUE THREAD

**Issue by KrisThielemans** [jaeandersson/swig#48]
At present,  SWIG_Error calls mexWarnMsgIdAndTxt.  SWIG_exception calls SWIG_Error and then SWIG_fail. The result of this is that when the wrapped code throws an exception, you get a MATLAB warning (with the exception text), and then an error (with the text 'Fatal error'). This is not good in my opinion. First instance, when catching the error in matlab, you still get the warning first.

I think SWIG_Error should call  mexErrMsgIdAndTxt instead. This seems the desired behaviour for SWIG_exception etc, but maybe this is not appropriate for SWIG_Error as it is also used in the traits code (where there is an argument that prevents it from throwing). As I'm not sure about the traits stuff, I cannot suggest the best solution...

**Comment by jaeandersson** [comment]
I think @jgillis found a solution to this for CasADi (@jgillis - right?)

**Comment by jaeandersson** [comment]
We don't have this issue in CasADi since @jgillis was able to use the "matlabprepend" and "matlabappend" features to change:

``` matlab
      [varargout{1:max(1,nargout)}] = casadiMATLAB_wrap(713, varargin{:});
```

into

``` matlab
      try

      [varargout{1:max(1,nargout)}] = casadiMATLAB_wrap(713, varargin{:});

      catch err
        if (strcmp(err.identifier,'SWIG:OverloadError'))
          msg = [swig_typename_convertor_cpp2matlab(err.message) 'You have: ' strjoin(cellfun(@swig_typename_convertor_matlab2cpp,varargin,'UniformOutput',false),', ')];
          throwAsCaller(MException(err.identifier,msg));
        else
          rethrow(err);
        end
      end
```

This essentially solves the issue for us, but I don't think it's a nice solution so I wouldn't advocate pushing it the swig branch. I think it could be solved properly with #58 though.

For future discussion, use #58.
